docs/src/main/asciidoc/rest-client-reactive.adoc

```diff
@@ -0,0 +1,594 @@
+        return countriesService.getByNameAsUni(name);
+    }
+}
+----
+
+[TIP]
+.Mutiny
+====
+The previous snippet uses Mutiny reactive types, if you're not familiar with them, read the link:getting-started-reactive#mutiny[Getting Started with Reactive guide] first.
+====
+
+When returning a `Uni`, every _subscription_ invokes the remote service.
+It means you can re-send the request by re-subscribing on the `Uni`, or use a `retry` as follows:
+
+[source, java]
+----
+
+@Inject @RestClient CountriesResource service;
+
+// ...
+
+service.nameAsync("Greece")
+   .onFailure().retry().atMost(10);
+
+----
+
+If you use a `CompletionStage`, you would need to call the service's method to retry.
+This difference comes from the laziness aspect of Mutiny and its subscription protocol.
+More details about this can be found in https://smallrye.io/smallrye-mutiny/#_uni_and_multi[the Mutiny documentation].
+
+== Custom headers support
+The MicroProfile REST client allows amending request headers by registering a `ClientHeadersFactory` with the `@RegisterClientHeaders` annotation.
+
+Let's see it in action by adding a `@RegisterClientHeaders` annotation pointing to a `RequestUUIDHeaderFactory` class in our `CountriesService` REST interface:
+
+[source, java]
+----
+package org.acme.rest.client;
+
+import org.eclipse.microprofile.rest.client.inject.RegisterRestClient;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import java.util.concurrent.CompletionStage;
+import java.util.Set;
+
+@Path("/v2")
+@RegisterRestClient
+@RegisterClientHeaders(RequestUUIDHeaderFactory.class)
+public interface CountriesService {
+
+    @GET
+    @Path("/name/{name}")
+    @Produces("application/json")
+    Set<Country> getByName(@PathParam("name") String name);
+
+    @GET
+    @Path("/name/{name}")
+    @Produces("application/json")
+    CompletionStage<Set<Country>> getByNameAsync(@PathParam("name") String name);
+}
+----
+
+And the `RequestUUIDHeaderFactory` would look like:
+
+[source, java]
+----
+package org.acme.rest.client;
+
+import org.eclipse.microprofile.rest.client.ext.ClientHeadersFactory;
+import org.jboss.resteasy.specimpl.MultivaluedMapImpl;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.ws.rs.core.MultivaluedMap;
+import java.util.UUID;
+
+@ApplicationScoped
+public class RequestUUIDHeaderFactory implements ClientHeadersFactory {
+
+    @Override
+    public MultivaluedMap<String, String> update(MultivaluedMap<String, String> incomingHeaders, MultivaluedMap<String, String> clientOutgoingHeaders) {
+        MultivaluedMap<String, String> result = new MultivaluedMapImpl<>();
+        result.add("X-request-uuid", UUID.randomUUID().toString());
+        return result;
+    }
+}
+----
+
+Any CDI bean matching the declared factory class will be picked up and can of course benefit from the usual CDI infrastructure.
```

Comment:
It isn't really clear to me what this means - it kinda reads like I don't need to use `@RegisterClientHeaders` which isn't the case

Reply:
I'll fix the source and the copy